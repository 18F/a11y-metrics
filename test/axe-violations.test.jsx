// @flow

const {
  getAxeViolationStats,
  flattenViolations,
  uniqueWebpages
} = require('../lib/components/axe-violations');

const violation = kind => ({
  kind,
  helpUrl: `https://info/${kind}`,
  nodeCount: 1
});

describe('uniqueWebpages()', () => {
  it('filters out records w/ identical homepages', () => {
    const records /*: any */ = [
      {website: {homepage: 'a'}},
      {website: {homepage: 'b'}},
      {website: {homepage: 'a'}},
    ];
    expect(uniqueWebpages(records)).toEqual([
      {website: {homepage: 'a'}},
      {website: {homepage: 'b'}},
    ]);
  });
});

describe('getAxeViolationStats()', () => {
  it('matches snapshot', () => {
    expect(getAxeViolationStats([
      violation('foo'),
      violation('bar'),
      violation('foo'),
      violation('bar'),
      violation('baz'),
    ])).toMatchSnapshot();
  });
});
