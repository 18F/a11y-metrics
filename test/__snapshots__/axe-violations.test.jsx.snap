// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAxeViolationStats() matches snapshot 1`] = `
Array [
  Object {
    "count": 2,
    "helpUrl": "https://info/bar",
    "name": "bar",
  },
  Object {
    "count": 2,
    "helpUrl": "https://info/foo",
    "name": "foo",
  },
  Object {
    "count": 1,
    "helpUrl": "https://info/baz",
    "name": "baz",
  },
]
`;
